pdp-engine/src/python_engine.rs

```diff
@@ -0,0 +1,265 @@
+use crate::builder::{PDPEngineBuilder, Present};
+use crate::error::PDPError;
+use crate::runner::PDPRunner;
+use crate::{PDPEngine, health};
+
+use async_trait::async_trait;
+use reqwest::{Client, RequestBuilder, Response};
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio_util::sync::CancellationToken;
+use url::Url;
+
+/// `PDPPythonEngine` provides a high-level API for interacting with a PDP server.
+/// It contains both a running runner and a connected HTTP client.
+#[derive(Clone)]
+pub struct PDPPythonEngine {
+    pub runner: PDPRunner,
+    pub base_url: Url,
+    pub client: Client,
+    health_monitor_token: CancellationToken,
+}
+
+impl std::fmt::Debug for PDPPythonEngine {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PDPPythonEngine")
+            .field("runner", &self.runner)
+            .field("base_url", &self.base_url)
+            // Skip client and health_monitor_token as they don't implement Debug
+            .finish_non_exhaustive()
+    }
+}
+
+impl PDPPythonEngine {
+    pub(crate) async fn new(
```

Comment:
As I see it new is a constructor and shouldn't have any side effect such as running the health monitor and instead should have start and stop to explicitly mention you start and stop it

Reply:
This code has been removed